ui/src/pages/Questions/Detail/components/WriteAnswer/useWriteAnswer.tsx

```diff
@@ -0,0 +1,293 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { marked } from 'marked';
+
+import { usePromptWithUnload, useCaptchaModal } from '@/hooks';
+import { Modal } from '@/components';
+import {
+  AdminSettingsWrite,
+  FormDataType,
+  PostAnswerReq,
+} from '@/common/interface';
+import { postAnswer } from '@/services';
+import { guard, handleFormError, SaveDraft, storageExpires } from '@/utils';
+import { DRAFT_ANSWER_STORAGE_KEY } from '@/common/constants';
+import { writeSettingStore } from '@/stores';
+
+interface Props {
```

Comment:
I can reuse props defined at FC. I am not sure of the right approach. whether writeAnswer FC and webhook should have same props or can they be different etc.

I think. Having the same props with Union of prop requirements of both FC & webhook will be great.